script.module.inputstreamhelper/lib/inputstreamhelper.py

```diff
@@ -0,0 +1,556 @@
+        if arch in config.X86_MAP:
+            return config.X86_MAP[arch]
+        elif 'armv' in arch:
+            arm_arch = 'armv' + arch.split('v')[1][:-1]
+            return arm_arch
+
+        return arch
+
+    def _log(self, string):
+        """InputStream Helper log method."""
+        logging_prefix = '[{0}-{1}]'.format(self._addon.getAddonInfo('id'), self._addon.getAddonInfo('version'))
+        msg = '{0}: {1}'.format(logging_prefix, string)
+        xbmc.log(msg=msg, level=xbmc.LOGDEBUG)
+
+    def _parse_chromeos_offset(self, bin_path):
+        """Calculate the Chrome OS losetup start offset using fdisk/parted."""
+        if self._cmd_exists('fdisk'):
+            cmd = ['fdisk', bin_path, '-l']
+        else:  # parted
+            cmd = ['parted', '-s', bin_path, 'unit s print']
+        self._log('losetup calculation cmd: {0}'.format(cmd))
+
+        output = subprocess.check_output(cmd)
+        self._log('losetup calculation output: {0}'.format(output))
+        for line in output.splitlines():
+            partition_data = line.split()
+            if partition_data:
+                if partition_data[0] == '3' or '.bin3' in partition_data[0]:
+                    offset = int(partition_data[1].replace('s', ''))
+                    return str(offset * config.CHROMEOS_BLOCK_SIZE)
+
+        self._log('Failed to calculate losetup offset.')
+        return False
+
+    def _run_cmd(self, cmd):
+        try:
+            subprocess.check_output(cmd)
+            self._log('{0} cmd executed successfully.'.format(cmd))
+            return True
+        except subprocess.CalledProcessError, error:
+            self._log('cmd failed with output: {0}'.format(error.output))
+            return False
+
+    def _set_loop_dev(self):
+        """Set an unused loop device that's available for use."""
+        cmd = ['losetup', '-f']
+        self._loop_dev = subprocess.check_output(cmd).strip()
+        self._log('Found free loop device: {0}'.format(self._loop_dev))
+        return True
+
+    def _losetup(self, bin_path):
+        """Setup Chrome OS loop device."""
+        cmd = ['losetup', self._loop_dev, bin_path, '-o', self._parse_chromeos_offset(bin_path)]
+        success = self._run_cmd(cmd)
+        if success:
+            self._attached_loop_dev = True
+            return True
+        else:
+            return False
+
+    def _mnt_loop_dev(self):
+        """Mount loop device to self._mnt_path()"""
+        dialog = xbmcgui.Dialog()
+        cmd = ['mount', '-t', 'ext2', self._loop_dev, '-o', 'ro', self._mnt_path()]
+        if os.getuid() != 0 and self._cmd_exists('sudo'):  # ask for permission to wrap cmd in sudo
+            if dialog.yesno(self._language(30001), self._language(30030), yeslabel=self._language(30029), nolabel=self._language(30028)):
+                cmd.insert(0, 'sudo')
+            else:
+                self._log('User refused to give sudo permission.')
+
+        success = self._run_cmd(cmd)
```

Comment:
please drop sudo rights here (sudo -k)

Reply:
Do you mean run sudo -k after the command has been executed?

Reply:
yes, we really need to limit root access to the absolute minimum.
Getting by without root would be very much preferred, although I haven't found a way that works for me yet without jumping through hoops.

Reply:
Yeah, I also hoped for a way to do it without root but it's a no-go. :-/